ibis/expr/rules.py

```diff
@@ -424,11 +424,33 @@ def table(arg, *, schema=None, **kwargs):
 
 @validator
 def column_from(name, column, *, this):
-    if not isinstance(column, (str, int)):
-        raise com.IbisTypeError(
-            f"value must be an int or str, got {type(column).__name__}"
-        )
-    return getattr(this, name)[column]
+    if not hasattr(this, name):
+        raise com.IbisTypeError(f"Could not get table {name} from {this}")
+    table = getattr(this, name)
+
+    if isinstance(column, (str, int)):
+        return table[column]
+    elif isinstance(column, ir.AnyColumn):
+        if not column.has_name():
+            raise com.IbisTypeError(f"Passed column {column} has no name")
+
+        maybe_column = column.get_name()
+        try:
+            if column.equals(table[maybe_column]):
+                return column
+            else:
```

Comment:
actually this else looks superflous, e.g. either the columns are equal (in which case it returns) or they are not or an error is raised.

so can you rm L441-444, pass on L446 ? and then you have the raise in L450

Reply:
The raised error would be not helpful (`"value must be an int or str or AnyColumn, got "`)